reconcile/openshift_base.py

```diff
@@ -225,6 +226,10 @@ def apply(dry_run, oc_map, cluster, namespace, resource_type, resource,
         except (MetaDataAnnotationsTooLongApplyError,
                 UnsupportedMediaTypeError):
             oc.replace(namespace, annotated.toJSON())
+        except FieldIsImmutableError:
```

Comment:
should we check if `resource_type` is `Route`? this can happen to StatefulSets as well, and deleting those may lead to unintended outages.

Reply:
unintentional changes to Routes may lead to outages as well. maybe the author's intention behind immutable fields was exactly that. and perhaps that should be our strategy as well? to create a new Route in one MR and delete the old one in the 2nd? just throwing thoughts.

Reply:
unintentional changes? not sure we have sageguards for that anywhere else... this change has the same effect of renaming the Route while changing the host, which can be done in the same MR to app-interface... but with this, tenants don't have to learn that they need to rename the route. I'm ok with just doccumenting that as well, but then it would be nice to fail the MR check when only the host is updated, with no renaming.

Reply:
i've no problem with this PR as is. just wanted to raise awareness around (particularly) StatefulSets.

Reply:
Yes yes, no question on that. Fixing.